providers/k8s/resources/ingress.go

```diff
@@ -20,8 +20,10 @@ import (
 )
 
 type mqlK8sIngressInternal struct {
-	lock sync.Mutex
-	obj  *networkingv1.Ingress
+	lock   sync.Mutex
+	obj    *networkingv1.Ingress
+	objId  string
+	mqlK8s *mqlK8s
```

Comment:
I only need this to later access the `GetSecrets` func. Perhaps there is a nicer way of doing this. Ideas are very welcome.

Reply:
You can request resources on the fly, especially useful if they are static like in `k8s` case:

```go
	o, err := CreateResource(runtime, "k8s", map[string]*llx.RawData{})
	if err != nil {
		return nil, err
	}
	k8s := o.(*mqlK8s)
```

Reply:
Thanks for this. Works like a charm.